### PR TITLE
feat(onboarding): add a workspaces category to the getting started steps (AYC-000)

### PR DESCRIPTION
**Pull request by chrismahlkeui**
Adds an **Arbeitsbereiche** category to the getting started steps, ahead of the workspaces launch. Eight steps that teach a workspace as a context rather than a folder.

| # | Step | Spotlight |
| --- | --- | --- |
| 1 | Arbeitsbereich erstellen | "Arbeitsbereich hinzufügen" on `/workspaces` |
| 2 | Arbeitsbereich vor dem Senden wählen | workspace picker in a new chat |
| 3 | Alternativ: Chat im Arbeitsbereich starten | composer on the workspace page |
| 4 | Bestehenden Chat zuordnen | chat row menu in the sidebar |
| 5 | Anweisung für den Arbeitsbereich hinterlegen | "Anweisungen" tab |
| 6 | Wissenssammlung im Arbeitsbereich erstellen | "Wissen" tab |
| 7 | Fähigkeit im Arbeitsbereich erstellen | "Fähigkeiten" tab |
| 8 | Arbeitsbereich zu Favoriten hinzufügen | star on the first workspace row |

Steps 2–8 depend on `createWorkspace` alone, so nothing is locked behind moving a chat. Step 2 is the everyday path and step 3 is marked as the alternative, so the list does not leave the reader choosing between two equal instructions.

## Copy decisions

- The two chat steps say only that the chat lands in the workspace. They do not promise instruction, knowledge or skills, because at that point in the sequence none of them exist yet — those promises live on the steps that create them.
- Steps 6 and 7 state that personal knowledge collections and skills stay active, since `ToolAssemblyService` merges them with the workspace's own (`mergeById`, `mergeKnowledgeBases`). Workspace-owned resources apply *in addition*, not instead.
- Favouriting comes last because it is a keep-this gesture, which only makes sense once the workspace has been used. It is also the only way a single workspace reaches the sidebar today.

## Feature toggle

The category is gated on `workspacesEnabled`. `useOnboardingProgress` only filtered on `adminOnly` before, so this adds toggle filtering — otherwise the steps would keep counting toward the totals and the sidebar progress while being unreachable.

## Spotlight handles

Each handle resolves to the rounded control so the highlight matches its shape. Where the control sits in normal flow the handle wraps it, as the existing targets do. Two controls cannot take a wrapper, so their handle sits on the control itself and `readSpotlightRadius` now prefers a handle that carries its own radius — a no-op for every existing target, whose handle is the rounding-free `OnboardingTourTarget` div:

- the sidebar chat menu is absolutely positioned, so a wrapper would collapse to zero height. It also drops `showOnHover` while a tour runs, so the spotlight lands on something visible.
- the favourites star sits behind the Tooltip's layout span in the shared `PinButton`, which takes the handle as a plain string since `shared` may not import from `widgets`.

The workspace picker carries the new chat page's 900ms entrance time on `data-tour-settle`, because joyride measures the target once and would otherwise land on the pre-animation position.

## Workflow classification

**Standard Path.** Frontend configuration, copy and UI wiring. No public contract, no persistence change (`UpdateOnboardingDto` validates step ids as free-form strings), no auth, tenant or migration surface. The one cross-cutting concern is the feature toggle, which has both states covered by tests.

## Validation

- `npx tsc --noEmit` clean
- `npx eslint --max-warnings=0` clean on every touched file
- `npx vitest run` — 176 files, 823 tests passing, rebased on current `main`
- Pre-commit hooks passed, including dependency-cruiser (it caught an FSD violation mid-work: the handle originally imported `widgets/onboarding` from `shared/ui/pin-button`)
- New tests: both toggle states (category visible/hidden, steps dropping out of totals and completion count) and the five navigation paths including the no-workspace fallback

The toggle contract is proven by test rather than a second instance: the frontend reads `workspacesEnabled` from the API response at runtime, not from `process.env` at boot, so both states are reachable without a restart.

## Review rounds

Three rounds of Cursor Bugbot findings, all valid and all fixed:

1. The sidebar spotlight had no target when the chats group is collapsed or holds no unpinned chat. The step now checks for the handle before navigating and omits the spotlight rather than opening a tour on a target that will never be found.
2. Two `data-tour="create-workspace"` nodes existed in the empty state, so the spotlight landed on the small header control instead of the empty-state action. Exactly one control per branch carries the handle now.
3. The instruction tab unmounted its content while loading, so its handle was absent right after navigation. The textarea now stays mounted and disabled, matching the knowledge and skills tabs.
4. An empty `workspaces` list was read as "none exist" even while the query was in flight, so a step clicked early got the create-workspace fallback. The fallback now waits for the query to settle.

`addDocuments` carries the same in-flight race against its knowledge-base query on `main`. It is left untouched here rather than widened into this PR.

## Open

A help center page for workspaces — if one exists, step 1 could take a `secondaryAction` to it the way "Wissenssammlung erstellen" does.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Low Risk**
> Frontend-only onboarding, navigation, and spotlight wiring behind `workspacesEnabled`; no API or auth changes.
> 
> **Overview**
> Adds a **Workspaces** getting-started category (eight steps) gated on `workspacesEnabled`, with EN/DE copy for the category and each step’s spotlight text.
> 
> **Onboarding behavior** now routes workspace steps through `handleWorkspaceAction`: deep-links to the first workspace and the right tab for instruction/knowledge/skills, navigates to `/chat` for picker and assign-chat flows, falls back to create-workspace when the list is empty (after load settles), and skips spotlights when there is no sidebar chat row or no unfavorited workspace.
> 
> **UI tour hooks** (`data-tour` / `OnboardingTourTarget`) land on create workspace, favourite pin (first unfavorited row via `findUnfavoritedWorkspace`), new-chat workspace picker (with settle delay), chat composer, sidebar chat menu (visible during tour), and workspace tab create actions. Instructions tab keeps the tour target mounted while loading; `TourRenderer` prefers radius on handles that already have rounding.
> 
> **Progress** filters categories by `requiresFeature` so disabled workspaces steps do not affect totals or completion. Shared `WORKSPACES_PER_PAGE` moves to the workspaces feature module. Tests cover navigation edge cases, favourite targeting, instructions loading, and toggle gating.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 5bb1047bf7b4f040d18ea32fbdd6b138364d51cf. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by chrismahlkeui** [comment]
* **#1709** <a href="https://app.graphite.com/github/pr/ayunis-core/ayunis-core/1709?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/ayunis-core/ayunis-core/1709?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: ayunis-core-frontend/src/widgets/app-sidebar/ui/ChatsSidebarGroup.tsx [hunk 1]
```diff
@@ -179,13 +180,18 @@ export function ChatsSidebarGroup() {
...
                     key={thread.id}
                     thread={thread}
                     isPinned={pinnedThreadIds.has(thread.id)}
                     onRename={handleRenameClick}
                     onDelete={handleDeleteClick}
+                    tourTarget={
+                      index === 0
+                        ? TOUR_TARGET.assignChatToWorkspace
+                        : undefined
+                    }
```
[the remark this answers is not in the document]
Comment (chrismahlke-ai) [reply ayunis-core-frontend/src/widgets/app-sidebar/ui/ChatsSidebarGroup.tsx]: Skipped rather than fixed as reported: with no unpinned chat, or with the chats group collapsed, there is no meaningful alternative target — rerouting to `createWorkspace` would not help a user who has no chat to assign. The step now checks for the sidebar handle before navigating and simply omits the spotlight when it is absent, so the tour no longer opens on a target joyride will never find. The step text already names the full path, so it stays completable either way.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: ayunis-core-frontend/src/pages/workspaces/ui/WorkspacesPage.tsx [hunk 2]
```diff
@@ -26,9 +27,11 @@ export default function WorkspacesPage({
...
 
   const createButton = (
-    <Button size="sm" onClick={() => setIsCreateOpen(true)}>
-      {t('page.addWorkspace')}
-    </Button>
+    <OnboardingTourTarget name={TOUR_TARGET.createWorkspace}>
+      <Button size="sm" onClick={() => setIsCreateOpen(true)}>
+        {t('page.addWorkspace')}
+      </Button>
+    </OnboardingTourTarget>
```
[the remark this answers is not in the document]
Comment (chrismahlke-ai) [reply ayunis-core-frontend/src/pages/workspaces/ui/WorkspacesPage.tsx]: Fixed. `createButton` is now a plain button and only the prominent control per branch is wrapped: the empty-state action when there are no workspaces, the header action otherwise. Exactly one `data-tour="create-workspace"` node exists in either branch, so the spotlight can no longer land on the small header control while the user is looking at the empty-state call to action.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: ayunis-core-frontend/src/pages/workspace/ui/WorkspaceInstructionsTab.tsx [hunk 3]
```diff
@@ -36,16 +37,18 @@ export function WorkspaceInstructionsTab({
...
+          value={value}
+          data-testid="workspace-instruction-input"
+          onChange={(event) => {
+            setValue(event.target.value);
+            setIsDirty(true);
+          }}
+          placeholder={t('context.instructions.placeholder')}
+          rows={8}
+        />
+      </OnboardingTourTarget>
```
[the remark this answers is not in the document]
Comment (chrismahlke-ai) [reply ayunis-core-frontend/src/pages/workspace/ui/WorkspaceInstructionsTab.tsx]: Fixed. The tab no longer unmounts its content while the context loads: the textarea stays rendered and disabled, matching the knowledge and skills tabs, which render their create button regardless of loading state. The handle is therefore present from the first frame after navigation. A failed request still replaces the section with the load error, which is the intended outcome.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: ayunis-core-frontend/src/pages/onboarding/ui/OnboardingStepItem.tsx [hunk 4]
```diff
@@ -120,6 +153,43 @@ export default function OnboardingStepItem({
...
+  const handleWorkspaceAction = (to: string, spotlight?: TourTargetName) => {
+    armReturn();
+
+    if (!firstWorkspace) {
+      void navigate({ to: '/workspaces' }).then(() =>
+        triggerSpotlight(TOUR_TARGET.createWorkspace, {
+          translationKey: 'createWorkspace',
+        }),
+      );
+      return;
```
[the remark this answers is not in the document]
Comment (chrismahlke-ai) [reply ayunis-core-frontend/src/pages/onboarding/ui/OnboardingStepItem.tsx]: Fixed. `useWorkspaces` now also hands back its loading state, and the create-workspace fallback only fires once the query has settled — an empty list means "none yet" only then, and in flight it means "not known". A step clicked mid-flight lands on the workspace overview without being told to create something that already exists. Worth noting that `addDocuments` carries the same race against its knowledge-base query on main; that one is untouched here rather than widened into this PR.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: ayunis-core-frontend/src/widgets/app-sidebar/ui/ChatsSidebarGroup.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (chrismahlkeui) [reply ayunis-core-frontend/src/widgets/app-sidebar/ui/ChatsSidebarGroup.tsx:194]: Skipped rather than fixed as reported: with no unpinned chat, or with the chats group collapsed, there is no meaningful alternative target — rerouting to `createWorkspace` would not help a user who has no chat to assign. The step now checks for the sidebar handle before navigating and simply omits the spotlight when it is absent, so the tour no longer opens on a target joyride will never find. The step text already names the full path, so it stays completable either way.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: ayunis-core-frontend/src/pages/workspaces/ui/WorkspacesPage.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (chrismahlkeui) [reply ayunis-core-frontend/src/pages/workspaces/ui/WorkspacesPage.tsx:39]: Fixed. `createButton` is now a plain button and only the prominent control per branch is wrapped: the empty-state action when there are no workspaces, the header action otherwise. Exactly one `data-tour="create-workspace"` node exists in either branch, so the spotlight can no longer land on the small header control while the user is looking at the empty-state call to action.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: ayunis-core-frontend/src/pages/workspace/ui/WorkspaceInstructionsTab.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (chrismahlkeui) [reply ayunis-core-frontend/src/pages/workspace/ui/WorkspaceInstructionsTab.tsx:55]: Fixed. The tab no longer unmounts its content while the context loads: the textarea stays rendered and disabled, matching the knowledge and skills tabs, which render their create button regardless of loading state. The handle is therefore present from the first frame after navigation. A failed request still replaces the section with the load error, which is the intended outcome.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: ayunis-core-frontend/src/pages/onboarding/ui/OnboardingStepItem.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (chrismahlkeui) [reply ayunis-core-frontend/src/pages/onboarding/ui/OnboardingStepItem.tsx:175]: Fixed. `useWorkspaces` now also hands back its loading state, and the create-workspace fallback only fires once the query has settled — an empty list means "none yet" only then, and in flight it means "not known". A step clicked mid-flight lands on the workspace overview without being told to create something that already exists. Worth noting that `addDocuments` carries the same race against its knowledge-base query on main; that one is untouched here rather than widened into this PR.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: ayunis-core-frontend/src/pages/onboarding/ui/OnboardingStepItem.tsx [hunk 5]
```diff
@@ -120,6 +153,47 @@ export default function OnboardingStepItem({
...
+      // An empty list means "none yet" only once the query has settled; while
+      // it is in flight it means "not known", so land on the overview without
+      // telling a user who already has a workspace to create one.
+      void navigate({ to: '/workspaces' }).then(() => {
+        if (areWorkspacesLoading) return;
+        triggerSpotlight(TOUR_TARGET.createWorkspace, {
+          translationKey: 'createWorkspace',
+        });
+      });
+      return;
```
[the remark this answers is not in the document]
Comment (chrismahlkeui) [reply ayunis-core-frontend/src/pages/onboarding/ui/OnboardingStepItem.tsx:175]: Fixed. The fallback now also stands down when the workspaces query settled with an error, not just while it is in flight, so a failed fetch no longer tells a user who has workspaces to create one. Covered by a test alongside the loading case.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: ayunis-core-frontend/src/pages/workspaces/ui/WorkspacesContent.tsx [hunk 6]
```diff
@@ -10,8 +10,12 @@ export function WorkspacesContent({
...
   return (
     <div className="flex flex-col gap-2">
-      {workspaces.map((workspace) => (
-        <WorkspaceRow key={workspace.id} workspace={workspace} />
+      {workspaces.map((workspace, index) => (
+        <WorkspaceRow
+          key={workspace.id}
+          workspace={workspace}
+          pinTourTarget={index === 0}
+        />
```
[the remark this answers is not in the document]
Comment (chrismahlkeui) [reply ayunis-core-frontend/src/pages/workspaces/ui/WorkspacesContent.tsx:27]: Fixed. The handle no longer sits on row zero unconditionally: `WorkspacesContent` now picks the first workspace that is *not* yet a favourite, mirroring how `SkillsPage` anchors its pin step on the first eligible skill rather than the first row. The spotlight can therefore no longer land on a control whose click would unpin. If every workspace is already favourited, no row carries the handle and the step simply shows no highlight, which is the harmless case since the step is then already done. Covered by three tests.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: ayunis-core-frontend/src/pages/onboarding/ui/OnboardingStepItem.tsx [hunk 7]
```diff
@@ -120,6 +157,40 @@ export default function OnboardingStepItem({
...
+    }
+
+    const spotlightAfterNavigation = () => {
+      if (spotlight) triggerSpotlight(spotlight);
+    };
+
+    if (!WORKSPACE_DETAIL_STEP_IDS.has(step.id)) {
+      void navigate({ to }).then(spotlightAfterNavigation);
+      return;
```
[the remark this answers is not in the document]
Comment (chrismahlkeui) [reply ayunis-core-frontend/src/pages/onboarding/ui/OnboardingStepItem.tsx:194]: Fixed, and at the root rather than one side of it. The target and the decision to launch lived in two components with no shared rule, which is why the previous fix left this gap. Both now call the same `findUnfavoritedWorkspace` helper in `features/workspaces`: `WorkspacesContent` uses it to place the handle, and the step uses it to decide whether to launch at all, so they cannot drift apart. With every workspace already favourited, the step navigates without starting a tour instead of leaving joyride waiting for a target that will never render.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: ayunis-core-frontend/src/pages/onboarding/ui/OnboardingStepItem.tsx [hunk 8]
```diff
@@ -120,6 +161,45 @@ export default function OnboardingStepItem({
...
+          translationKey: 'createWorkspace',
+        });
+      });
+      return;
+    }
+
+    const hasNoTarget =
+      (spotlight === TOUR_TARGET.assignChatToWorkspace &&
+        isSidebarSpotlightMissing(spotlight)) ||
+      (spotlight === TOUR_TARGET.favoriteWorkspace && !hasUnfavoritedWorkspace);
```
[the remark this answers is not in the document]
Comment (chrismahlkeui) [reply ayunis-core-frontend/src/pages/onboarding/ui/OnboardingStepItem.tsx:181]: Fixed. The page size is now a single exported constant that the route and the helper share, and the helper only considers that first page, so the check and the rendered rows can no longer disagree.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: ayunis-core-frontend/src/features/workspaces/index.ts [hunk 9]
```diff
@@ -2,3 +2,7 @@ export { useWorkspace } from './api/useWorkspace';
 export { useWorkspaces } from './api/useWorkspaces';
 export { useDeleteWorkspace } from './api/useDeleteWorkspace';
 export type { Workspace } from './model/types';
+export {
+  findUnfavoritedWorkspace,
+  WORKSPACES_PER_PAGE,
+} from './lib/findUnfavoritedWorkspace';
```
[the remark this answers is not in the document]
Comment (chrismahlkeui) [reply ayunis-core-frontend/src/features/workspaces/index.ts:8]: Fixed. The `workspaces/` paragraph in `features/SUMMARY.md` now lists `findUnfavoritedWorkspace` and `WORKSPACES_PER_PAGE` in `lib/`, and says why they are shared: the overview places the favourite tour handle with that rule and the getting-started step decides from the same rule whether to launch the spotlight, so the two cannot drift apart.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: ayunis-core-frontend/src/pages/onboarding/ui/OnboardingStepItem.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (chrismahlkeui) [reply ayunis-core-frontend/src/pages/onboarding/ui/OnboardingStepItem.tsx:181]: Fixed on both sides, the same way the in-flight workspaces query is already handled: an empty favourites list only counts as "nothing favourited" once the query has settled. While it is loading, `WorkspacesContent` places no handle and the step launches no tour, so the highlight can no longer land on a star that would unpin. Covered by a test in each component.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

**Review by sirolad — APPROVED** [review]
